src/components/Library/Dropdown.js

```diff
@@ -29,3 +29,9 @@ export function DropdownItem({children, onClick, style}) {
     <a onClick={onClick} style={style} className="dropdown-item">{children}</a>
```

Comment:
So your dropdown component current looks like this:
```jsx
export default function Dropdown({trigger, children, align}) {
  return(
    <div className="dropdown is-hoverable" >
      <div className="dropdown-trigger">
          <span className="icon is-small">
            <i className="fas fa-angle-down" aria-hidden="true"></i>
          </span>
      </div>
      <div className="dropdown-menu" id="dropdown-menu" role="menu">
        <div className="dropdown-content">
          {children}
        </div>
      </div>
    </div>
  )
}
```

You declare three props used by your component: `trigger`, `children` and `align`. You currently only use `children`.
The dropdown component is composed of a trigger and a hidden menu. When you click on the trigger, it reveals the hidden menu containing the dropdown items with the links.
The dropdown component is one of those components which is a container and does not know beforehand what it will contain: what links the menu will display and what element will serve as the trigger.
With the component `DropdownItem` you are now able to display your own links from external components, like so:
```jsx
<Dropdown>
  <DropdownItem onClick={() => onEdit(todo)}>Edit</DropdownItem>
  <DropdownItem onClick={() => onDelete(todo.id)}>
    <span className="has-text-danger">Delete</span>
  </DropdownItem>
  ...
</Dropdown>
```

But now, we want to be able to also have a custom content for the trigger. Imagine the classic example of an app with the profile menu dropdown in the top right of the screen: we don't have a simple caret like in our curent component but rather the name of the user and an avatar as the trigger text.

**So we want to pass another `React.Node` from the parent component to the Dropdown**. The "problem" is that you already pass a `React.Node` from the parent component through the prop named `children`.
See the part of the dropdown component:
```jsx
<div className="dropdown-content">
  {children}
</div>
```

The prop "children" is the built-in way to pass content to a child component, but React also allows you to pass content through any prop you like.
So what we can do is add a trigger property that we will insert inside the ".dropdown-trigger" div.
Your component would then look like this:
```jsx
export default function Dropdown({ trigger, children }) {
  return(
    <div className="dropdown is-hoverable" >
      <div className="dropdown-trigger">
        {trigger}
      </div>
      <div className="dropdown-menu" id="dropdown-menu" role="menu">
        <div className="dropdown-content">
          {children}
        </div>
      </div>
    </div>
  )
}
```

Then, to be iso (have exactly the same view as before) with what we currently have, you would need to use the dropdown this way:
```jsx
<Dropdown
  trigger={(
    <span className="icon is-small">
      <i className="fas fa-angle-down" aria-hidden="true" />
    </span>
  )}
>
  <DropdownItem onClick={() => onEdit(todo)}>Edit</DropdownItem>
  <DropdownItem onClick={() => onDelete(todo.id)}>
    <span className="has-text-danger">Delete</span>
  </DropdownItem>
  ...
</Dropdown>
```

Reply:
Once we are able to pass whatever trigger to our dropdown component, you can then do the last instructions of the doc

Reply:
aaah omg i actually did that but forgot the () in the trigger props in the todo component -> facepalm.
Thanks haha